tft_bot/helpers/screen_helpers.py

```diff
@@ -1,221 +1,145 @@
-        x1,y1 (int): The coordinates of the top left of the region to be searched.
-        x2,y2 (int): The coordinates of the bottom right of the region to be searched.
-        precision (float, optional): The precision to be used when matching the image. Defaults to 0.8.
+        path: The relative or absolute path to the image to be found.
+        precision: The precision to be used when matching the image. Defaults to 0.8.
+        offsets: A tuple of coordinates to off-set the region by. Useful if you only want to check a specific region.
+          All offsets are from the top-left of the game window.
 
     Returns:
-        (bool | list[int] | tuple[int, int]): The coordinates of the image if found within the specified region,
-            None otherwise.
+        The position of the image and it's width and height or None if it wasn't found
     """
-    try:
-        image = resource_path(image)
-        pos = imagesearch.imagesearcharea(image, x1, y1, x2, y2, precision)
-        count = 0
-
-        while pos[0] == -1:
-            time.sleep(timesample)
-            pos = imagesearch.imagesearcharea(image, x1, y1, x2, y2, precision)
-            count = count + 1
-            if count > max_samples:
-                break
-        return pos
-    except Exception:
-        return None
+    return get_on_screen(
+        window_title=CONSTANTS["window_titles"]["game"], path=path, precision=precision, offsets=offsets
+    )
 
 
-def find_image(path: str, precision: float = 0.8) -> None | list[int] | tuple[int, int]:
-    """Search for a given image, returning the coordinates if found.
+def get_on_screen(
+    window_title: str, path: str, precision: float = 0.8, offsets: tuple[int, int, int, int] | None = None
+) -> tuple[int, int, int, int] | None:
+    """
+    Check if a given image is detected on screen in a specific window's area.
 
     Args:
-        path (str): The relative or absolute path to the image to be found.
-        precision (float, optional): The precision to be used when matching the image. Defaults to 0.8.
+        window_title: The title of the window we should look at.
+        path: The relative or absolute path to the image to be found.
+        precision: The precision to be used when matching the image. Defaults to 0.8.
+        offsets: A tuple of coordinates to off-set the region by. Useful if you only want to check a specific region.
+          All offsets are from the top-left of the game window.
 
     Returns:
-        (None | list[int] | tuple[int, int]): The coordinates of the image if found within the specified region,
-            None otherwise.
+        The position of the image and it's width and height or None if it wasn't found
     """
-    try:
-        path = resource_path(path)
-        pos = imagesearch.imagesearch(path, precision)
-        return pos if pos[0] != -1 else None
-    except Exception:
+    window_bounding_box = get_window_bounding_box(window_title=window_title)
```

Comment:
One thing I see us doing a lot of is re-getting the same `get_window_bounding_box` with the same methods, etc. I wonder if it makes sense to implement a re-use for this?

A couple of options coming to mind could be:
- Implement the re-use at the "user" level (the callers), which would probably be a good chunk more work and may be less readable
- Implement a "caching" in the helpers that is using something akin to a hash (whether it's a `dict` or something properly typed) matching `window_title` to the result, and pulling from that?
  - If we go this approach, due to the user being able to manipulate the screen or something possibly having changed, we'd likely want to implement a "cap" of how many sequential uses it can have before it gets busted from the cache and gets re-looked up (like maybe every 10 uses, for example), or some other way to clear the cache so it doesn't infinitely re-use this (maybe a function in the core loop calls a sub-method in this module that clears the cache each iteration).


This is definitely not strictly necessary, just something that looks like a slightly more "expensive" computation and could slow this down a bit.

Reply:
I also thought of this and ran a quick performance test. Getting the handle takes around 0.0006ms, and getting the rectangle of the window about the same, so it is not worth just from a performance standpoint. We could follow the holy book of OOP and make a class out of the screen_helper (i.e., `class ScreenReader`), but I was too lazy lol

Reply:
Ahh gotcha, if it's truly that little performance impact, that works fine for me and is not worth the additional overhead. We can revisit this if it's an issue in the future!